main/integration-tests/__test__/api-tests/authentication/get-auth-provider-types.test.js

```diff
@@ -0,0 +1,386 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Get authentication types list scenarios', () => {
+  let setup;
+  let adminSession;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('Get authentication types list', () => {
+    it('should fail if user is not an admin', async () => {
+      const researcherSession = await setup.createResearcherSession();
+
+      await expect(researcherSession.resources.authentication.types().get()).rejects.toMatchObject({
+        code: errorCode.http.code.forbidden,
+      });
+    });
+
+    it('should fail if user is inactive', async () => {
+      const admin2Session = await setup.createAdminSession();
+      await adminSession.resources.users.deactivateUser(admin2Session.user);
+
+      await expect(admin2Session.resources.authentication.types().get()).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should list auth provider types', async () => {
+      const admin2Session = await setup.createAdminSession();
+
+      await expect(admin2Session.resources.authentication.types().get()).resolves.toEqual(
+        expect.arrayContaining([
+          {
```

Comment:
Same as other comment, what do you think of taking this into another file, the file returns a function that you can call and pass it whatever parameters you need, it should return this json object.

Reply:
These are the only two built-in provider types that will always be returned for this API call. The function would have to return this list, regardless of the parameters, unless more provider types are added in the future. 

Let me know if I'm misunderstanding your comment.

Reply:
Moving the large JSON bodies in different files, per discussion.